op-node/rollup/derive/channel_out.go

```diff
@@ -70,33 +70,32 @@ func (co *ChannelOut) Reset() error {
 	return nil
 }
 
-// AddBlock adds a block to the channel. It returns an error
-// if there is a problem adding the block. The only sentinel
-// error that it returns is ErrTooManyRLPBytes. If this error
-// is returned, the channel should be closed and a new one
-// should be made.
-func (co *ChannelOut) AddBlock(block *types.Block) error {
+// AddBlock adds a block to the channel. It returns the RLP encoded byte size
+// and an error if there is a problem adding the block. The only sentinel error
+// that it returns is ErrTooManyRLPBytes. If this error is returned, the channel
+// should be closed and a new one should be made.
+func (co *ChannelOut) AddBlock(block *types.Block) (uint64, error) {
```

Comment:
Thanks, added `ChannelOut.InputBytes() int`!

This allowed me to remove the duplicated tracking of total input bytes. Because of that, now I don't need the returned `rlpbytes` of `ChannelOut.AddBlock` any more. Should I revert returning them? I can still imagine scenarios in the future where it could come in handy to have these returned, and it would be in line with some io functions of the std lib, like `io.Copy`, that return the written amount of data. No strong opinion on my side, but I'd keep it.

Reply:
I'm fine either way. If you lean to keeping it, go for it